src/plugins/intel_cpu/src/emitters/snippets/aarch64/jit_memory_emitters.cpp

```diff
@@ -132,31 +132,34 @@ void jit_memory_emitter::emit_code_impl(const std::vector<size_t>& in_idxs,
     auto reg_runtime_params = dnnl::impl::cpu::aarch64::abi_param1;
     XReg aux_gpr = is_offset_runtime ? XReg(static_cast<int>(aux_gpr_idxs.back())) : XReg(0);
 
-    XReg data_reg(0);
-    if (in_out_type_ == emitter_in_out_map::gpr_to_vec) {
-        data_reg = XReg(in_idxs[0]);
-    } else if (in_out_type_ == emitter_in_out_map::vec_to_gpr) {
-        data_reg = XReg(out_idxs[0]);
-    } else {
-        OV_CPU_JIT_EMITTER_THROW("unsupported in_out_type");
-    }
+    std::vector<size_t> eff_in = in_idxs;
+    std::vector<size_t> eff_out = out_idxs;
 
     if (is_offset_runtime) {
+        XReg data_reg(0);
+        if (in_out_type_ == emitter_in_out_map::gpr_to_vec) {
+            data_reg = XReg(in_idxs[0]);
+        } else if (in_out_type_ == emitter_in_out_map::vec_to_gpr) {
+            data_reg = XReg(out_idxs[0]);
+        } else {
+            OV_CPU_JIT_EMITTER_THROW("unsupported in_out_type");
+        }
+
         // load the runtime offset from args.buffer_offsets[buffer_cluster_id]
         h->ldr(aux_gpr,
                ptr(reg_runtime_params,
                    static_cast<int32_t>(GET_OFF(buffer_offsets) + buffer_cluster_id * sizeof(size_t))));
-        // bump the pointer
-        h->add(data_reg, data_reg, aux_gpr);
-    }
-
-    emit_impl(in_idxs, out_idxs);
+        h->add(h->X_DEFAULT_ADDR, data_reg, aux_gpr);
```

Comment:
As far as I understand, we can use `X_DEFAULT_ADDR` only if we can guarantee that it will not be corrupted/overlapped by the child code. In this particular case we definitely can't guarantee that, since `emit_impl` may potentially have the code which use `X_DEFAULT_ADDR` as well.
To avoid hardly debugable issues, I would propose to avoid `X_DEFAULT_ADDR` usage, and choose a more safe way: take an additional register from `aux_gpr_idxs`

Reply:
Discussed offline, it seems to be reasonable tradeoff. Applied the comment